giskard/project.py

```diff
@@ -346,7 +347,6 @@ def _validate_label_with_target(classification_labels, target_values=None):
                 raise ValueError(f"Target column value {invalid_target_values} not declared in "
                                  f"classification_labels list: {classification_labels}")
 
-
     @staticmethod
     def _validate_classification_labels(classification_labels, model_type):
```

Comment:
Why do you need to explicitly call `str(label)` on line 360?
It looks a bit weird that this validation method outputs classification_labels and not just validates it

Reply:
This part of code is actually untouched but thanks for bringing it to notice. We can optimise this. I think we call str(label) because initially the classification_labels were more to explain to business users and were preferred to be in string until AI Test came into the picture.  This is more for the front end preparation for AI Inspect.

Reply:
Just to add to this conversation. We still do prefer to be in string for better readability but currently, if user is sending numeric values, we are converting them to string. This is the first place of the implementation and the second one is for target values. I think its a good idea to remove this conversation if its not essential for AI Inspect.

Reply:
on this note, I also have a question about 
```
target_values = target_values if isinstance(target_values, str) else [str(label) for label in target_values]
```

In which case do we expect `target_values` to be a string?

Reply:
We expect the target_values to be string for better readability on inspect, but even if its not, we change it to string as we compare the target with the classification_labels which is always a string. This change is done especially for this scenario: 
classification_labels given by user : [1,2]
In giskard we convert it to string so we send classification_labels = ["1", "2"]
target values in dataset =  [1,2]

if we dont convert this to string, we have an error .